File: spec/python/test_type_ternary_opaque.py
import unittest
from type_ternary_opaque import TypeTernaryOpaque

class TestTypeTernaryOpaque(unittest.TestCase):
    def test_type_ternary_opaque(self):
        with TypeTernaryOpaque.from_file('src/term_strz.bin') as r:
            self.assertEqual(r.dif.s1, "foo")
            self.assertEqual(r.dif.s2, "bar")
            self.assertEqual(r.dif.s3, "|baz@")

File: spec/lua/test_type_ternary_opaque.lua
local luaunit = require("luaunit")

require("type_ternary_opaque")

TestTypeTernaryOpaque = {}

function TestTypeTernaryOpaque:test_type_ternary_opaque()
    local r = TypeTernaryOpaque:from_file("src/term_strz.bin")

    luaunit.assertEquals(r.dif.s1, "foo")
    luaunit.assertEquals(r.dif.s2, "bar")
    luaunit.assertEquals(r.dif.s3, "|baz@")
end
